src/libp2p_crypto.erl

```diff
@@ -260,6 +319,31 @@ bin_to_pubkey(Network, <<NetType:4, ?KEYTYPE_ED25519:4, PubKey:32/binary>>) ->
     case NetType == from_network(Network) of
         true -> {ed25519, PubKey};
         false -> erlang:error({bad_network, NetType})
+    end;
+bin_to_pubkey(
+    Network,
+    <<
+        NetType:4,
+        ?KEYTYPE_MULTISIG:4,
+        M:?MULTISIG_KEY_INDEX_BITS/integer-unsigned-little,
+        N:?MULTISIG_KEY_INDEX_BITS/integer-unsigned-little,
+        KeysDigest/binary
+    >>
+) ->
+    case NetType == from_network(Network) of
+        true ->
+            case M =< N of
+                true ->
+                    case multihash:hash(KeysDigest) of
+                        {error, Reason} ->
+                            erlang:error({bad_multihash, Reason});
```

Comment:
Why are we throwing instead of returning an error? Confirming that theses are these supposed to be unexpected errors, like in other places errors are thrown in this lib?

Reply:
Bin to pubkey also throws on invalid network type above?

Reply:
This API here does not have a tagged return, so we can only throw to indicate misuse.